src/signature/spki.rs

```diff
@@ -472,154 +449,139 @@ mod tests {
         Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
-    test_verify_signed_data!(test_rsa_pkcs1_sha1_wrong_algorithm,
-                             "rsa-pkcs1-sha1-wrong-algorithm.pem",
-                             Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
     // XXX: RSA PKCS#1 with SHA-1 is a supported algorithm, but we only accept
     // 2048-8192 bit keys, and this test file is using a 1024 bit key. Thus,
     // our results differ from Chromium's. TODO: this means we need a 2048+ bit
     // version of this test.
-    test_verify_signed_data!(test_rsa_pkcs1_sha1, "rsa-pkcs1-sha1.pem",
-                             Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+    test_verify_signature!(test_rsa_pkcs1_sha1,
+                           "rsa-pkcs1-sha1.pem",
+                           &spki::RSA_PKCS1_2048_8192_SHA1,
+                           Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
     // XXX: RSA PKCS#1 with SHA-1 is a supported algorithm, but we only accept
     // 2048-8192 bit keys, and this test file is using a 1024 bit key. Thus,
     // our results differ from Chromium's. TODO: this means we need a 2048+ bit
     // version of this test.
-    test_verify_signed_data!(test_rsa_pkcs1_sha256, "rsa-pkcs1-sha256.pem",
-                             Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+    test_verify_signature!(test_rsa_pkcs1_sha256,
+                           "rsa-pkcs1-sha256.pem",
+                           &spki::RSA_PKCS1_2048_8192_SHA256,
+                           Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+    // TODO this only tests DER decoding, not signature logic
     test_parse_spki_bad_outer!(test_rsa_pkcs1_sha256_key_encoded_ber,
                                "rsa-pkcs1-sha256-key-encoded-ber.pem",
                                VerifyWithSPKIError::BadDER);
-    test_verify_signed_data!(test_rsa_pkcs1_sha256_spki_non_null_params,
+    test_verify_signature!(test_rsa_pkcs1_sha256_spki_non_null_params,
                              "rsa-pkcs1-sha256-spki-non-null-params.pem",
+                             &spki::RSA_PKCS1_2048_8192_SHA256,
                              Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
-    test_verify_signed_data!(
-        test_rsa_pkcs1_sha256_using_ecdsa_algorithm,
-        "rsa-pkcs1-sha256-using-ecdsa-algorithm.pem",
-        Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
-    test_verify_signed_data!(
+    test_verify_signature!(
         test_rsa_pkcs1_sha256_using_id_ea_rsa,
         "rsa-pkcs1-sha256-using-id-ea-rsa.pem",
+        &spki::RSA_PKCS1_2048_8192_SHA256,
         Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
 
-    // Chromium's PSS test are for parameter combinations we don't support.
-    test_verify_signed_data!(test_rsa_pss_sha1_salt20_using_pss_key_no_params,
```

Comment:
PSS tests deleted: no point in a bunch more tests that only check that 1024 bit keys aren't supported...